server/auth/registrar.go

```diff
@@ -25,7 +30,344 @@ var (
+		if err != nil {
+			log.Warnf("Error sending postbond result to user %v: %v", acctID, err)
+			if err = auth.Send(acctID, resp); err != nil {
+				log.Warnf("Error sending postbond result to account %v: %v", acctID, err)
+				// The user will need to 'connect' to reconcile bond status.
+			}
+		}
+		return nil
+	}
+
+	// See if the account exists, and get known unexpired bonds. Also see if the
+	// account has previously paid a legacy registration fee.
+	dbAcct, bonds, _, legacyFeePaid := auth.storage.Account(acctID, lockTimeThresh)
+	newAcct := dbAcct == nil
+
+	reqConfs := int64(auth.bondConfs)
+
+	// See if we already have this bond in DB.
+	for _, bond := range bonds {
+		if bond.AssetID == assetID && bytes.Equal(bond.CoinID, bondCoinID) {
+			postBondRes.Active = true
+			_, postBondRes.Tier = auth.AcctStatus(acctID)
+			return sendResp()
+		}
+	}
+
+	bondStr := coinIDString(assetID, bondCoinID)
+	bondAssetSym := dex.BipIDSymbol(assetID)
+
+	// Determine if it exists and is confirmed.
+	chkAddr, chkAmt, confs, err := auth.checkFee(assetID, bondCoinID)
+	if err != nil {
+		// Not found is expected if this is a pre-broadcast validation request.
+		if !errors.Is(err, asset.CoinNotFoundError) {
+			return msgjson.NewError(msgjson.BondError, "unable to check bond with node: %v", err)
+		}
+
+		// OK. Respond with active=false. They may now broadcast, wait for the
+		// required confirms, and postbond again.
+		log.Debugf("Validated prospective bond txn output %s (%s) paying %d for user %v",
+			bondStr, bondAssetSym, amt, acctID)
+		postBondRes.Active = false
+		return sendResp()
+	}
+
+	if int64(chkAmt) != amt {
+		return msgjson.NewError(msgjson.BondError, "node reports incorrect bond amount: %d != %d", chkAmt, amt)
+	}
+
+	if chkAddr != bondAddr {
+		return msgjson.NewError(msgjson.BondError, "node reports incorrect bond address: %v != %v", chkAddr, bondAddr)
+	}
+
+	dbBond := &db.Bond{
+		Version:  postBond.BondVer,
+		AssetID:  assetID,
+		CoinID:   bondCoinID,
+		Script:   postBond.BondScript,
+		Amount:   amt,
+		LockTime: lockTime,
+	}
+
+	// Either store the bond or start a block waiter to activate the bond and
+	// respond with a PostBondResult when it is fully-confirmed.
+	bondIDKey := bondKey(assetID, bondCoinID)
+	if !auth.registerBondWaiter(bondIDKey) {
+		// Waiter already running! They'll get a response to their first
+		// request, or find out on connect if the bond was activated.
+		return msgjson.NewError(msgjson.BondAlreadyConfirmingError, "bond already submitted")
+	}
+
+	var refundAddr string
+	if legacyFeePaid {
+		refundAddr = postBond.LegacyFeeRefundAddr
+	}
+
+	if confs >= reqConfs {
+		// No need to call checkFee again in a waiter.
+		postBondRes.Active = true
+		auth.storeBondAndRespond(conn, dbBond, acct, newAcct, msg.ID, postBondRes, refundAddr)
+		auth.removeBondWaiter(bondIDKey) // after storing it
+		return nil
+	}
+
+	auth.latencyQ.Wait(&wait.Waiter{
+		Expiration: time.Now().Add(time.Hour),
```

Comment:
Need to change this to a few minutes since this is now just network latency we're accounting for here.